core/src/main/java/io/questdb/griffin/engine/ops/InsertOperationImpl.java

```diff
@@ -87,6 +90,27 @@ public OperationFuture execute(SqlExecutionContext sqlExecutionContext) throws S
         }
     }
 
+    public ObjList<CharSequence> getColumnNames() {
+        return columnNames;
+    }
+
+    public TableToken getTableToken() {
+        return tableToken;
+    }
+
+    public void setColumnNames(ObjList<CharSequence> columnNameList) {
+        if (columnNames == null) {
+            if (columnNameList.size() == 0) {
+                columnNames = EMPTY_COLUMN_LIST;
+            } else {
+                columnNames = new ObjList<>(columnNameList);
+            }
+        } else {
+            columnNames.clear();
+            columnNames.addAll(columnNameList);
```

Comment:
can it happen that this is called multiple times?
just thinking if it can happen ever that first `columnNames` is set to `EMPTY_COLUMN_LIST`, then next time we come in, `columnNames` is not null anymore, so we add some columns to it on the else branch.

or is it guaranteed that if `columnNameList` was empty first, then it will be always empty?

Reply:
It shouldn't be called more than once on the same InsertOperation but I'll add some aditional checks to detect if it happens in the future . Thanks for pointing it out.